File: Others/PG_로또의최고순위와최저순위.py
def solution(lottos, win_nums):
    rank = [6, 5, 4, 3, 2]
    answer = []
    chk = zeros = 0
    for num in lottos:
        if num in win_nums:
            chk += 1
            continue
        if num == 0:
            zeros += 1

    for i in range(len(rank)):
        if rank[i] == (zeros + chk):
            answer.append(i + 1)
        if rank[i] == chk:
            answer.append(i + 1)
    if len(answer) == 1:
        answer.append(6)
    if len(answer) == 0:
        answer = [6, 6]

    return answer

File: Others/PG_로또의최고순위와최저순위.js
// 내 풀이
function solution(lottos, win_nums) {
  let answer = [];
  let cnt = 0
  let match = 0
  for (let n of lottos) {
      if (n === 0) {
          cnt++
      } else {
          for (let num of win_nums) {
              if (n === num) {
                  match++
                  break
              }
          }
      }
  }
  if (cnt + match >= 2) {
      answer.push(7-(match+cnt))
      match >= 2 ? answer.push(7-match) : answer.push(6)
  } else {
      answer = [6, 6]
  }
  return answer;
}


// 좋은 풀이
// array의 find와 map을 적절히 잘 사용했다!!
function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  let answer = [],
      ans = [],
      ans1 = [];

  lottos.map(x => {
      let val = win_nums.find(y => y == x); // 같은 값이 존재하면 val은 값이 아니면 val은 undefined
      
      if(x == 0) ans1.push(x)
      // undefined는 false이다!
      if(val) {
          ans.push(val)
          ans1.push(val)
      }
  })
  answer.push(rank[ans1.length])
  answer.push(rank[ans.length])

  return answer;
}